### PR TITLE
feat: add signTx param to sign transaction as optional

### PR DESCRIPTION
**Pull request by alexruzenhack**
### Acceptance Criteria
- add param signTx to prepareTransaction in src/utils/transaction.ts
- add optional attribute to prepareCreateNewToken in src/new/wallet.ts
- add optional attribute to prepareMintTokensData in src/new/wallet.ts
- add optional attribute to prepareMeltTokensData in src/new/wallet.ts

### Security Checklist
- [x] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

**Review by r4mmer — APPROVED** [review]
We just need to make sure we don't use these methods in atomic-swap since the wallet-service facade assumes that all inputs are from our wallet.

File: src/wallet/wallet.ts [hunk 1]
```diff
@@ -1282,21 +1284,24 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
 
     const tx = new CreateTokenTransaction(name, symbol, inputsObj, outputsObj);
 
-    const dataToSignHash = tx.getDataToSignHash();
+    // Sign transaction
+    if(newOptions.signTx) {
```
Comment (pedroferreira1) [inline src/wallet/wallet.ts:1288]:
```suggestion
    if (newOptions.signTx) {
```

File: src/wallet/wallet.ts [hunk 2]
```diff
@@ -1650,23 +1663,27 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
 
     const tx = new Transaction(inputsObj, outputsObj);
     tx.tokens = [token];
-    const dataToSignHash = tx.getDataToSignHash();
 
-    if (!newOptions.pinCode) {
-      throw new Error('PIN not specified in prepareMeltTokensData options');
-    }
+    // Sign transaction
+    if(newOptions.signTx) {
```
Comment (pedroferreira1) [inline src/wallet/wallet.ts:1668]:
```suggestion
    if (newOptions.signTx) {
```

File: src/wallet/wallet.ts (hunk 1 above)
Comment (alexruzenhack) [reply src/wallet/wallet.ts]: 061a41b2f67c0e31f2eab0c5e43816b3e3466dab

File: src/wallet/wallet.ts (hunk 2 above)
Comment (alexruzenhack) [reply src/wallet/wallet.ts]: Done: 061a41b2f67c0e31f2eab0c5e43816b3e3466dab

File: src/utils/transaction.ts [hunk 3]
```diff
@@ -449,12 +449,15 @@ const transaction = {
    * @param tx tx data to be prepared
    * @param pinCode pin to unlock the mainKey for signatures
    * @param storage Storage to get the mainKey
+   * @param signTx sign transaction instance (default true)
    * @returns {Promise<Transaction>} Prepared transaction
    */
-  async prepareTransaction(txData: IDataTx, pinCode: string, storage: IStorage): Promise<Transaction> {
+  async prepareTransaction(txData: IDataTx, pinCode: string, storage: IStorage, signTx = true): Promise<Transaction> {
```
[the remark this answers is not in the document]
Comment (alexruzenhack) [reply src/utils/transaction.ts]: Done: 061a41b2f67c0e31f2eab0c5e43816b3e3466dab

**Review by pedroferreira1 — APPROVED** [review]

**Pull request merged**